### ISSUE TITLE
not overriding image environment variables by kubernetes

### ISSUE THREAD

**Issue by wozniakjan** [kubernetes-incubator/cri-o#1293]
Logging team has tested the cri-o support and there is one issue that currently prevents us from using cri-o, @mrunalp asked me to create an issue here.

**Description**

If an image is built using docker and provides an environment variable, kubernetes should be able to override the same environment variable. Currently, this behavior differs.

- default for an environment variable "ES_HOST=localhost" is in the [image](https://github.com/openshift/origin-aggregated-logging/blob/master/curator/Dockerfile.centos7#L8)
- kubernetes DC [overrides](https://github.com/openshift/openshift-ansible/blob/master/roles/openshift_logging_curator/templates/curator.j2#L68-L69) the value
- the override is not correctly propagated to the environment of the container

**Steps to reproduce the issue:**

Define an environment variable in dockerfile and try to override it in pod yaml

```Dockerfile
FROM centos:centos7
ENV testvar=dockerfile
CMD echo [$testvar] ; sleep infinity
```

```yaml
kind: Pod
apiVersion: v1
metadata:
  name: override-has
spec:
  containers:
    - name: has 
      image: docker.io/wozniakjan/crio_has
      env:
        - name: testvar
          value: kubernetes_yaml
```

Other tested combinations work well:
https://gist.github.com/wozniakjan/c9e342359c7d9c2d96db0e8a1bc5280a

**Describe the results you received:**
`$testvar` has value `[dockerfile]`

**Describe the results you expected:**
`$testvar` should have value `[kubernetes_yaml]`

**Output of `crio --version`:**

```
crio version 1.0.6
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
```
openshift_deployment_type=origin
openshift_use_crio=true
openshift_use_crio_only=true
openshift_crio_systemcontainer_image_override=registry.access.redhat.com/openshift3/cri-o:v3.7
openshift_disable_check=docker_storage,memory_availability,disk_availability
```

original bz for reference: https://bugzilla.redhat.com/show_bug.cgi?id=1517605

**Comment by runcom** [comment]
Looking...

**Comment by jcantrill** [comment]
@runcom Is there an update to this issue?

**Comment by runcom** [comment]
I'm looking into it right now

**Comment by runcom** [comment]
It looks like the cri-o version you're using is from the 1.7 kube release cycle so it might be missing a backport actually.

**Comment by runcom** [comment]
This is fixed in latest CRI-O 1.0.8 and later versions (1.8.x, 1.9.x, master) by https://github.com/kubernetes-incubator/cri-o/pull/1186 - could you test it out guys?

**Comment by wozniakjan** [comment]
confirming it works with 1.9.1, also confirming that latest 1.7 still didn't get the backport. Thanks for your help!
